sympy/ntheory/generate.py

```diff
@@ -505,8 +512,8 @@ def eval(cls, n):
         return S(arr2[1])
 
 
-def nextprime(n, ith=1):
-    """ Return the ith prime greater than n.
+class nextprime(Function):
```

Comment:
I do not know why this does not work now. I think it will be better if we can somehow make this work rather than changing the test.

Reply:
Well, that's because you made it a class.
.you have to do something like this 
```
  assert nextprime(x).subs(x, 15) == 17
```
as you've done for the testing module.

Reply:
It will break existing API. I am in favor of making subclass of `Function` support this API.

Reply:
> It will break existing API. I am in favor of making subclass of `Function` support this API.

Good Idea

Reply:
@asmeurer what do you think about this?